### ISSUE TITLE
Wrap `Contour.jl` as a backend for `polygonize`

### ISSUE THREAD

**Issue by asinghvi17** [asinghvi17/GeometryOps.jl#81]
This would be neat to have as an extension package, and could even be used to tie into Rasters etc for handling if GeoInterface ever gets a raster API.

Contour.jl is used in the Makie `contour` recipe.

**Comment by rafaqz** [comment]
Yeah there are a few kinds of `polygonise`... square, angular and smoothed curves. We should have all of them. Im not sure what the algs are called.
